zebra/zebra_rib.c

```diff
@@ -1619,6 +1620,11 @@ static bool rib_compare_routes(const struct route_entry *re1,
 	 * v6 link-locals, and we also support multiple addresses in the same
 	 * subnet on a single interface.
 	 */
+	if ((re1->type == ZEBRA_ROUTE_CONNECT &&
+	     re2->type == ZEBRA_ROUTE_CONNECT) &&
```

Comment:
we already tested that the two types are equal - can just test either re1 or re2?

Reply:
good point, fixed